app/Models/Themes.php

```diff
@@ -1,124 +1,223 @@
+			$iterator->next();
 		}
+
+		sort($list);
+
 		return $list;
 	}
 
-	public static function get_infos($theme_id) {
-		$theme_dir = PUBLIC_PATH . self::$themesUrl . $theme_id;
-		if (is_dir($theme_dir)) {
-			$json_filename = $theme_dir . '/metadata.json';
-			if (file_exists($json_filename)) {
-				$content = file_get_contents($json_filename);
-				$res = json_decode($content, true);
-				if ($res &&
-						!empty($res['name']) &&
-						isset($res['files']) &&
-						is_array($res['files'])) {
-					$res['id'] = $theme_id;
-					return $res;
+	/**
+	 * @return bool
+	 */
+	public static function isAvailable(string $theme) {
+		return in_array($theme, self::getNames());
+	}
+
+	/**
+	 * Extract metadata for the selected theme
+	 *
+	 * @return array
+	 */
+	private static function extractMetadata(SplFileInfo $file, string $current_user) {
+		$metadata = "{$file->getPathName()}/metadata.json";
+
+		if (!file_exists($metadata) || !is_readable($metadata)) {
+			return [];
+		}
+
+		if (false === $content = file_get_contents($metadata)) {
+			return [];
+		}
+
+		$metadata = json_decode($content, true);
+		if (JSON_ERROR_NONE !== json_last_error()) {
+			return [];
+		}
+
+		$public = null === $current_user ? "/themes/{$metadata['name']}" : "/themes/{$current_user}/{$metadata['name']}";
+		$metadata['public'] = $public;
+		$metadata['thumbnail'] = "{$public}/thumbs/original.png";
+
+		$iconpath = PUBLIC_PATH . "{$public}/icons";
+		if (file_exists($iconpath)) {
+			$metadata['iconpath'] = "{$public}/icons";
+			$iterator = new FilesystemIterator($iconpath);
+			$iterator->rewind();
+			while ($iterator->valid()) {
+				$item = $iterator->current();
+				if ($item->isFile()) {
+					$metadata['icons'][$item->getFilename()] = $item->getFilename();
 				}
+				$iterator->next();
 			}
 		}
-		return false;
+
+		return $metadata;
 	}
 
-	private static $themeIconsUrl;
-	private static $themeIcons;
+	/**
+	 * @return array|false
+	 */
+	public static function load(string $theme) {
+		$current_user = Minz_Session::param('currentUser');
 
-	public static function load($theme_id) {
-		$infos = self::get_infos($theme_id);
-		if (!$infos) {
-			if ($theme_id !== self::$defaultTheme) {	//Fall-back to default theme
-				return self::load(self::$defaultTheme);
-			}
-			$themes_list = self::getList();
-			if (!empty($themes_list)) {
-				if ($theme_id !== $themes_list[0]) {	//Fall-back to first theme
-					return self::load($themes_list[0]);
-				}
-			}
+		// Load user theme
+		$metadata = self::extractMetadata(new SplFileInfo(THEME_PATH . "/{$current_user}/{$theme}"), $current_user);
+
+		// Load user default theme
+		$defaultTheme = self::$defaultTheme;
+		if (empty($metadata)) {
+			$metadata = self::extractMetadata(new SplFileInfo(THEME_PATH . "/{$current_user}/{$defaultTheme}"), $current_user);
+		}
+
+		// Load system default theme
+		if (empty($metadata)) {
+			$metadata = self::extractMetadata(new SplFileInfo(THEME_PATH . "/{$defaultTheme}"), null);
```

Comment:
```
PHP Fatal error:  Uncaught TypeError: FreshRSS_Themes::extractMetadata():
Argument #2 ($current_user) must be of type string, null given,
called in /var/www/FreshRSS/app/Models/Themes.php on line 165
and defined in /var/www/FreshRSS/app/Models/Themes.php:111
Stack trace:
#0 /var/www/FreshRSS/app/Models/Themes.php(165): FreshRSS_Themes::extractMetadata()
#1 /var/www/FreshRSS/app/FreshRSS.php(104): FreshRSS_Themes::load()
#2 /var/www/FreshRSS/app/FreshRSS.php(142): FreshRSS::loadStylesAndScripts()
#3 /var/www/FreshRSS/app/layout/layout.phtml(1): FreshRSS::preLayout()
#4 /var/www/FreshRSS/lib/Minz/View.php(88): include('...')
#5 /var/www/FreshRSS/lib/Minz/View.php(101): Minz_View->includeFile()
#6 /var/www/FreshRSS/lib/Minz/View.php(68): Minz_View->buildLayout()
#7 /var/www/FreshRSS/lib/Minz/Dispatcher.php(54): Minz_View->build()
#8 /var/www/FreshRSS/lib/Minz/FrontController.php(81): Minz_Dispatcher->run()
#9 /var/www/FreshRSS/p/i/index.php(70): Minz_FrontController->run()
#10 {main}
  thrown in /var/www/FreshRSS/app/Models/Themes.php on line 111
```

Reply:
:/ I've never experienced that. Under what conditions do you have this error?

Reply:
I've found the error!

Reply:
It's fixed

Reply:
Tested in PHP8